docs/studios/unity/unity-sdk/csharp/configuration.mdx

```diff
@@ -0,0 +1,128 @@
+  - `BotEntryPoint.cs` which extends `RGUserBot.cs` and defines the main C# script entry points for your bot used by the Regression Games SDK to launch and communicate with the bot.
+    -  By convention, the namespace of this file and all other `.cs` files for your bot should match the folder name of the bot.  This is important so that different bots' `BotEntryPoint` classes are in unique namespaces.
+
+You may add as many additional scripts or dependencies to your bot as necessary, but the 2 files above MUST exist for your bot to work correctly.
+
+## Bot Configuration
+
+Your Unity Integration allows flexibility in how your GameObjects are set up, spawned into the scene, and torn down.
+The first step to writing your C# Bot is defining how these processes should handle this Bot.
+
+The `ConfigureBot(RG rgObject)` method accepts an `RG` object which is provided by Regression Games.
+This object exposes configuration options (defined in the sections below) and methods for describing Bot behaviors and validations.
+Implementing the `ConfigureBot` method is optional, but should be implemented when you want to configure your bot before processing the first tick.
+
+For spawnable Bots, `rgObject.CharacterConfig` is an object containing character configurations to help seat your Bot as the appropriate class/prefab/etc.
+This is a freeform object to be interpreted by your game's implementation of the
+[`RGBotSpawnManager.SeatBot`](../RGBotSpawnManager#public-virtual-void-seatbotbotinformation-bottospawn)
+and [`RGSpawnManager.SpawnBot`](../RGBotSpawnManager#canbenull-public-abstract-gameobject-spawnbotbool-latejoin-botinformation-botinformation) interfaces.
+
+<Tabs>
+<TabItem value="character_config_simple" label="Simple definition" default>
+
+```javascript
+public override void ConfigureBot(RG rgObject)
+{
+    rgObject.CharacterConfig = new Dictionary<string, object>() {
+        { "type", "Mage" }
+    };
+}
+```
+
+</TabItem>
+<TabItem value="character_type_freeform" label="Complex definition" default>
+
+```javascript
+
+public override void ConfigureBot(RG rgObject)
+{
+    rgObject.CharacterConfig = new Dictionary<string, object>() {
+        { "role", "Warrior" },
+        { "specialization", "Reaver" },
+        { "startingEquipment", new Dictionary<string, object>()
+            {
+                { "mainHand", "Sword" },
+                { "offHand", "Sword" }
+            }
+        },
+        { "spawn", "BlueTower" },
+        { "maxHP", 50 },
+        { "immunities", new string[] { "fire" } }
+    };
+}
+```
+
+</TabItem>
+</Tabs>
+
+### `IsSpawnable`
+
+Specifies whether this Bot should be represented by a visible GameObject.
+Bots that simulate players or NPCs are generally spawnable, while Bots that interact with menus, 
+represent environmental hazards, observe the game state without interaction, etc. are not.
+
+```javascript 
+protected override bool GetIsSpawnable()
```

Comment:
At some point I'd like to see all the configs being set on the `rgObject` so that they match `characterConfig` and imitate the javascript configurations.

Right now the statement at the beginning of this section ("The `ConfigureBot(RG rgObject)` method accepts an `RG` object which is provided by Regression Games.
This object exposes configuration options (defined in the sections below) and methods for describing Bot behaviors and validations.") doesn't make much sense because the only configuration that is currently set this way is `characterConfig`